crates/orchestrator/src/jobs/state_update_job/mod.rs

```diff
@@ -31,26 +53,207 @@ impl Job for StateUpdateJob {
+        let metadata_tx_hashes = job
+            .metadata
+            .get(&format!("{}{}", JOB_METADATA_STATE_UPDATE_ATTEMPT_PREFIX, attempt_no))
+            .expect("Could not find tx hashes metadata for the current attempt")
+            .clone()
+            .replace(' ', "");
+
+        let tx_hashes: Vec<&str> = metadata_tx_hashes.split(',').collect();
+        let block_numbers = self.get_block_numbers_from_metadata(job)?;
+        let settlement_client = config.settlement_client();
+
+        for (tx_hash, block_no) in tx_hashes.iter().zip(block_numbers.iter()) {
+            let tx_inclusion_status = settlement_client.verify_tx_inclusion(tx_hash).await?;
+            match tx_inclusion_status {
+                SettlementVerificationStatus::Rejected(_) => {
+                    job.metadata.insert(JOB_METADATA_STATE_UPDATE_LAST_FAILED_BLOCK_NO.into(), block_no.to_string());
+                    return Ok(tx_inclusion_status.into());
+                }
+                // If the tx is still pending, we wait for it to be finalized and check again the status.
+                SettlementVerificationStatus::Pending => {
+                    settlement_client.wait_for_tx_finality(tx_hash).await?;
+                    let new_status = settlement_client.verify_tx_inclusion(tx_hash).await?;
+                    match new_status {
+                        SettlementVerificationStatus::Rejected(_) => {
+                            job.metadata
+                                .insert(JOB_METADATA_STATE_UPDATE_LAST_FAILED_BLOCK_NO.into(), block_no.to_string());
+                            return Ok(new_status.into());
+                        }
+                        SettlementVerificationStatus::Pending => {
+                            return Err(eyre!("Tx {tx_hash} should not be pending."))
+                        }
+                        SettlementVerificationStatus::Verified => {}
+                    }
+                }
+                SettlementVerificationStatus::Verified => {}
+            }
+        }
+        // verify that the last settled block is indeed the one we expect to be
+        let expected_last_block_number = block_numbers.last().expect("Block numbers list should not be empty.");
+        let out_last_block_number = settlement_client.get_last_settled_block().await?;
+        let block_status = if out_last_block_number == *expected_last_block_number {
+            SettlementVerificationStatus::Verified
+        } else {
+            SettlementVerificationStatus::Rejected(format!(
+                "Last settle bock expected was {} but found {}",
+                expected_last_block_number, out_last_block_number
+            ))
+        };
+        Ok(block_status.into())
     }
 
     fn max_process_attempts(&self) -> u64 {
-        todo!()
+        5
     }
 
     fn max_verification_attempts(&self) -> u64 {
-        todo!()
+        1
```

Comment:
let's make it like 10? so we check till 10mis?

Reply:
Done